protocols/pop3/src/main/java/org/apache/james/protocols/pop3/core/POP3MessageCommandDelegate.java

```diff
@@ -46,24 +47,25 @@ public POP3MessageCommandDelegate(Collection<String> commands) {
     public Response handleMessageRequest(POP3Session session, Request request) {
         if (session.getHandlerState() == POP3Session.TRANSACTION) {
 
-            POP3MessageCommandArguments args = POP3MessageCommandArguments.fromRequest(request);
-            if (!args.isValid()) {
+            Optional<POP3MessageCommandArguments> args = POP3MessageCommandArguments.fromRequest(request);
+            if (args.isEmpty()) {
                 return handleSyntaxError();
             }
-
+            int num = args.get().getMessageNumber();
```

Comment:
Calling `.get` on Optionals is cheatting.

You are supposed to transform it, and unbox it in a functional style. Method extraction helps a lot. Eg:

```
return args.map(this::handleValidArgs).orElseGet(this::handleSyntaxError);
```

Reply:
Right, looks a lot cleaner this way. Fixed in latest push.